File: test/fixtures/stdlib/pathlib_extended.py

from pytra.utils.assertions import py_assert_all, py_assert_eq
from pathlib import Path


def run_pathlib_extended() -> bool:
    root = Path("work/transpile/obj/pathlib_case32")
    root.mkdir(parents=True, exist_ok=True)

    child = root / "values.txt"
    child.write_text("42")

    checks: list[bool] = []
    checks.append(py_assert_eq(child.exists(), True, "exists"))
    checks.append(py_assert_eq(child.name, "values.txt", "name"))
    checks.append(py_assert_eq(child.stem, "values", "stem"))
    checks.append(py_assert_eq((child.parent / "values.txt").exists(), True, "parent_join_exists"))
    checks.append(py_assert_eq(child.read_text(), "42", "read_text"))
    return py_assert_all(checks, "pathlib_extended")

if __name__ == "__main__":
    print(run_pathlib_extended())

File: test/fixtures/stdlib/pathlib_extended.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Any = System.Object;
using int64 = System.Int64;
using float64 = System.Double;
using str = System.String;
using Pytra.CsModule;
using Path = Pytra.CsModule.py_path;

public static class Program
{
    public static bool run_pathlib_extended()
    {
        Path root = new Path("work/transpile/obj/pathlib_case32");
        root.mkdir(true, true);
        
        Path child = System.Convert.ToDouble(root) / System.Convert.ToDouble("values.txt");
        child.write_text("42");
        
        System.Collections.Generic.List<bool> checks = new System.Collections.Generic.List<bool>();
        checks.Add(System.Object.Equals(child.exists(), true));
        checks.Add(System.Object.Equals(child.name(), "values.txt"));
        checks.Add(System.Object.Equals(child.stem(), "values"));
        checks.Add(System.Object.Equals(System.Convert.ToDouble(child.parent()) / System.Convert.ToDouble("values.txt").exists(), true));
        checks.Add(System.Object.Equals(child.read_text(), "42"));
        return System.Linq.Enumerable.All(checks, __x => System.Convert.ToBoolean(__x));
    }
    
    public static void Main(string[] args)
    {
            System.Console.WriteLine(run_pathlib_extended());
    }
}
